### PR TITLE
Add user parameter to projects.

### PR DESCRIPTION
**Pull request by LyonesGamer**
fixes issue #289

## Changes in this PR.

- [X] Test the parameter
- [X] Implement the parameter

## Testing this PR.

1. Run ``npm test``.
2. Create a testing instance of TimeSync, then (after authenticating) make a request to ``/projects?user=tschuy``.

### Expected Output.

1. The two new tests should pass.
2. You should see a list only of the projects to which tschuy belongs.

@osuosl/devs

### CODE REVIEW

**Comment by LyonesGamer** [comment]
On a side note, I think that we should refactor the times tests to look more like this. One list of all times, then use filter() and map() to get the expected results for a given test, instead of redefining the whole list every time. It'll make it easier to maintain and reduce unnecessary redundancy.

**Comment by ElijahCaine** [comment]
These tests look good. To address the side note I would *not* refactor the time tests unless you find yourself working on the time tests anyway.

**Comment by LyonesGamer** [comment]
Agreed.

**Comment by LyonesGamer** [comment]
@ElijahCaine I could use some help. I've left some console.log statements so you can see what I'm talking about, but basically my code is hitting the resolve() call inside of a promise, but never entering the then() function on that promise (Ctrl+F ``console.log('here')`` and ``console.log('there')``).

File: tests/projects.js [hunk 1]
```diff
@@ -198,6 +198,47 @@ module.exports = function(expect, request, baseUrl) {
     });
   });
 
+  describe('GET /projects?user=:username', function() {
+    it('returns all times for a user', function(done) {
```
Comment (Kennric) [inline tests/projects.js]: you mean "returns all projects for a user"?

**Comment by Kennric** [comment]
Another note for a tests refactor - we should really replace user/project/activity names with generic names like user1, user2,  project1, project2, (or valid_user_1, invalid_project, etc) rather than have gwm and tschuy all over our test data. A recursive  /tschuy/user_1/ should take care of that pretty easily.

**Comment by ElijahCaine** [comment]
LGTM +1

**Comment by Kennric** [comment]
+1

**Pull request merged**